src/main/java/core/basesyntax/storage/Storage.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> inventory = new HashMap<>();
+
+    public void addFruit(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new IllegalArgumentException("Quantity to add cannot be negative: " + quantity);
+        }
+        inventory.merge(fruit, quantity, Integer::sum);
+    }
+
+    public void removeFruit(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new IllegalArgumentException("Quantity to remove cannot be negative: "
+                    + quantity);
+        }
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Insufficient quantity to remove: " + fruit);
+        }
+        inventory.merge(fruit, -quantity, Integer::sum);
+    }
+
+    public Map<String, Integer> getInventory() {
+        return new HashMap<>(inventory);
+    }
+
+    public int getQuantity(String fruit) {
+        return inventory.getOrDefault(fruit, 0);
+    }
```

Comment:
The class `Storage` is not following the encapsulation principle. The inventory should be hidden from direct access. Consider making it private and providing necessary public methods to interact with it.